src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/NuGetVSActionTelemetryService.cs

```diff
@@ -24,6 +26,17 @@ public NuGetVSTelemetryService(ITelemetrySession telemetrySession)
         {
         }
 
+        public static void Initialize()
+        {
+            lock (LockObject)
```

Comment:
I noticed we were technically initializing the singleton multiple times so I used this `lock(object)` pattern.  Is there a better pattern I should do?  Is this even necessary in your opinion?

Reply:
The object is stateless, but I don't see the harm in keeping this.

Reply:
What are your thoughts on this pattern @zivkan ?

Reply:
As Nikolche said, the object is stateless, so even if two different threads create two different instances, one of them will get disposed, but no harm is done.

I'd personally remove the lock, rather than keeping as is. But it really won't make much of a difference either way, since there's so little code inside the lock.

Reply:
If you do keep the lock, I think:
```cs
if (condition)
{
  lock (_lockObject)
  {
    if (condition)
    {
      // do work
    }
  }
}
```

is higher performance, since it avoids needing to get the lock unnecesserily. But considering this should only happen at most twice in a single process, it's really not a significant concern.

Reply:
Okay I think I'll just skip the lock for now as I haven't found a scenario where initialization is run simultaneously.